### ISSUE TITLE
Only translate trait method when used

### ISSUE THREAD

**Issue by Nadrieril** [AeneasVerif/charon#487]
Currently, whenever we translate a trait like `Iterator`, we translate the signature of its many dozen methods. We could instead start each trait without any methods and only translate each method when we encounter it in the code.

**Comment by sonmarcho** [comment]
I agree with this. Just a small request: it would be good to have an option to switch between the two behaviors (but this is probably what you had in mind).

**Comment by Nadrieril** [comment]
I intend this to be part of the normal crate exploration behavior, therefore if you want to include all iterator methods you can `--include core::iter::Iterator::*`.

**Comment by sonmarcho** [comment]
I often use the current behavior to generate models for the backends: it's an easy of getting the trait declaration, method declarations and instance declarations for free. I'm perfectly fine if what you suggest becomes the new default behavior, but it would be good if we could use an option to *re-activate* the current behavior (should be simple right?).

**Comment by sonmarcho** [comment]
Following an offline discussion: the best solution so far is to have options which are actually sugar for `--include` of the proper pattern. This way we have a general mechanism to control which items get included or not and the implementation doesn't have to handle ad-hoc cases, while users have shortcuts which are easy to use for some common patterns.
